mutiny-core/src/nostr/mod.rs

```diff
@@ -1172,6 +1197,242 @@ impl<S: MutinyStorage> NostrManager<S> {
         Ok(event_id)
     }
 
+    /// Creates a recommendation event for a federation
+    pub async fn recommend_federation(
+        &self,
+        invite_code: &InviteCode,
+    ) -> Result<EventId, MutinyError> {
+        let kind = Kind::from(18173);
+        // first pull down our current recommendation event, we need to do this
+        // to make sure we don't overwrite any other recommendations
+        let filter = Filter::new().kind(kind).author(self.public_key).limit(1);
+        // fetch events
+        let events = self
+            .client
+            .get_events_of(vec![filter], Some(Duration::from_secs(5)))
+            .await?;
+
+        let current = events
+            .into_iter()
+            .find_map(|e| if e.kind == kind { Some(e) } else { None });
+
+        // tag the federation invite code
+        let invite_code_tag = Tag::Generic(
+            TagKind::U,
+            vec![invite_code.to_string(), "fedimint".to_string()],
+        );
+
+        // todo tag the federation announcement event, to do so we need to have the pubkey of the federation
```

Comment:
this isn't really possible?

Reply:
I'm a little unsure how this is supposed to work and link together. Is it just by the `d_tag` being the same across reviews? Does the federation invite code even need to be here? Should we pull down the other federation members invite code's to put in the vec?

Reply:
>Is it just by the d_tag being the same across reviews?

Yeah the d tag is the federation id

> Does the federation invite code even need to be here?

Yeah, in the case the federation has not announced itself

> Should we pull down the other federation members invite code's to put in the vec?

We could... I figured since it is a recommendation, they recommend the code they are using, maybe that is the most reliable member in the federation.